pkg/skaffold/yaml/yaml.go

```diff
@@ -55,3 +56,28 @@ func Marshal(in interface{}) (out []byte, err error) {
 	}
 	return b.Bytes(), nil
 }
+
+// MarshalWithSeparator is same as yaml.Marshal except for slice or array types where each element is encoded individually and separated by "---".
+func MarshalWithSeparator(in interface{}) (out []byte, err error) {
```

Comment:
if it's not too difficult, a unit test for this would be nice

Reply:
done